tests/integration/defs/perf/test_perf_sanity.py

```
@@ -2290,6 +2381,75 @@ def parse_metrics_from_output(output: str) -> Optional[Dict[str, float]]:
+        Preserves ``check_output`` semantics -- combined stdout/stderr returned
+        on success, ``CalledProcessError`` on a nonzero exit -- and adds the
+        bound that was missing: if the client neither finishes nor fails within
+        the budget, kill it and raise naming what it was waiting on, with the
+        partial client output and any server-side errors attached.
+
+        Without this the stage cannot fail on its own; only Slurm or Jenkins
+        stops it, hours later, with no results XML.
+        """
+        timeout_s = _benchmark_client_timeout()
+        started = time.monotonic()
+        proc = subprocess.Popen(cmd, env=env, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)